django/contrib/gis/gdal/geometries.py

```diff
@@ -546,16 +590,26 @@ def y(self):
     @property
     def z(self):
         "Return the Z coordinate for this Point."
-        if self.coord_dim == 3:
+        if self.is_3d:
             return capi.getz(self.ptr, 0)
 
+    @property
+    def m(self):
+        "Return the M coordinate for this Point."
+        if self.is_measured:
+            return capi.getm(self.ptr, 0)
+
     @property
     def tuple(self):
         "Return the tuple of this point."
         if self.coord_dim == 2:
             return (self.x, self.y)
         elif self.coord_dim == 3:
-            return (self.x, self.y, self.z)
+            if self.is_3d:
+                return (self.x, self.y, self.z)
+            return (self.x, self.y, self.m)
+        elif self.coord_dim == 4:
+            return (self.x, self.y, self.z, self.m)
```

Comment:
I'm wondering if this can be more straightforward? Maybe:

```python
        if self.is_3d and self.is_measured:
            return self.x, self.y, self.z, self.m
        if self.is_3d:
            return self.x, self.y, self.z
        if self.is_measured:
            return self.x, self.y, self.m
        return self.x, self.y
```

This drops the `== 2` branch as [`OGR_G_CoordinateDimension`](https://gdal.org/api/vector_c_api.html#_CPPv425OGR_G_CoordinateDimension12OGRGeometryH) can only return `2`, `3`, or `4` anyway.

We could replace `if self.is_3d and self.is_measured:` with `if self.coord_dim == 4:`, but this way is perhaps clearer in meaning.